src/compiler/scala/tools/nsc/transform/ExplicitOuter.scala

```diff
@@ -447,9 +447,10 @@ abstract class ExplicitOuter extends InfoTransform
           if (sym.isProtected && //(4)
               (qsym.isTrait || !(qual.isInstanceOf[Super] || (qsym isSubClass currentClass))))
             sym setFlag notPROTECTED
-          super.transform(tree)
+          if (sym.isStaticMember) tree
+          else super.transform(tree)
 
-        case Apply(sel @ Select(qual, nme.CONSTRUCTOR), args) if isInner(sel.symbol.owner) =>
+        case treeInfo.Applied(sel @ Select(qual, name), _, List(args)) if (name == nme.CONSTRUCTOR || sel.symbol.isStaticMember) && isInner(sel.symbol.owner) =>
```

Comment:
I'd change this to `sel.symbol.isClassConstructor || sel.symbol.isStaticMember` to consistently ask questions of the symbol.